### PR TITLE
Renderer enhancements

### PR DESCRIPTION
**Pull request by fritsch**
VDPAU: Fix obsolete setting referencing
LinuxRenderer: Use cheap methods when downscaling ... 

We would use Lanczos3 optimized when downscaling from 4k to 1080 which is heavy load and questionable concerning quality. As there is no setting to achieve this - disable it for now. Let's see if someone realizes.

### CODE REVIEW

File: xbmc/cores/VideoPlayer/VideoRenderers/LinuxRendererGL.cpp [hunk 1]
```diff
@@ -2862,6 +2862,10 @@ bool CLinuxRendererGL::Supports(ESCALINGMETHOD method)
     if (scaleX < minScale && scaleY < minScale)
       return false;
 
+    // avoid using hq scalers when downscaling
```
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/VideoRenderers/LinuxRendererGL.cpp]: I don't think that this is wanted for all cases. Assume I run Kodi in windowed mode on my laptop. Why do you want me to have lower quality in this case?

Comment (fritsch) [inline xbmc/cores/VideoPlayer/VideoRenderers/LinuxRendererGL.cpp]: The question is rather: what do you gain?

The input is let's say 3840x2160 and you want to display it on a 720x360 window. You really think that you will see the difference between lanczos3 optimized and bilinear downscaling?

**Comment by FernetMenta** [comment]
a much better strategy would be to detect this at runtime. if lots of drop/dups occur, reduce quality of scalers, de-interlacers, etc.

**Comment by fritsch** [comment]
Yes, that's another idea. I will check with RenderManager

Though if there is no real quality gain, e.g. people are watching on their desktop next to their real work ... I thought it would be okay.

**Comment by FernetMenta** [comment]
What about users not having a Full HD TV? They would suffer from lower quality when watching HD movies.

**Comment by fritsch** [comment]
The original setting was introduced so that we avoid a hq scaling algorithm when e.g. watching 1918x800 files on a 1920 display. Cause we knew that e.g. Lanczos3 costs too much when running on a 1920 surface. Now as 4k becomes the standard we run into that issue again. As input surface is 3840.

I will look into a method to attach to m_queueSkips but most likely this needs an interface change as Renderers are only asked with Supports(ScalingAlgorithm) method. They don't have a state concerning the skips. So RenderManager needs to keep the m_queueSkips and reset the scaling algorithm.

**Comment by fritsch** [comment]
This needs a big interface change to get a parameter for ```CLinuxRendererGL::UpdateVideoFilter``` and propagate this through all the method calls in WinRenderer, LinuxRenderer, VDPAURenderer and so on.

**Comment by FernetMenta** [comment]
if it needs an interface change, now it is the best time for it. this branch has already lots of interface changes. VideoPlayer::calcdroprequirement seem to be the best place to set those kind of features on demand. maybe change the name of the method to something more appropriate.

**Comment by fritsch** [comment]
That's a good place in deed. No need to change the method Name, we can call
a new method that will care for it. Will have a look the next days.

2015-09-06 21:50 GMT+02:00 Rainer Hochecker <notifications@github.com>:

> if it needs an interface change, now it is the best time for it. this
> branch has already lots of interface changes.
> VideoPlayer::calcdroprequirement seem to be the best place to set those
> kind of features on demand. maybe change the name of the method to
> something more appropriate.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/FernetMenta/xbmc/pull/271#issuecomment-138119019>.
>



-- 
                   Key-ID:     0x1A995A9B
                   keyserver: pgp.mit.edu
==============================================================
Fingerprint: 4606 DA19 EC2E 9A0B 0157  C81B DA07 CF63 1A99 5A9B

**Comment by fritsch** [comment]
Need to think over this - next free weekend I am coming :(

**Pull request closed without merging**